tests/sentry/api/endpoints/notifications/test_notification_actions_index.py

```diff
@@ -358,32 +392,30 @@ class MockActionRegistration(ActionRegistration):
         assert response.data["targetIdentifier"] == service["id"]
         assert response.data["targetDisplay"] == service["service_name"]
 
+    @patch("sentry.notifications.models.notificationaction.ActionRegistration.validate_action")
     @patch.dict(NotificationAction._registry, {})
-    def test_post_simple(self):
+    def test_post_simple(self, validate_action):
+
         class MockActionRegistration(ActionRegistration):
-            validate_action = MagicMock()
+            def fire(self, data: Any) -> None:
+                raise NotImplementedError
 
         registration = MockActionRegistration
-        NotificationAction.register_action(
-            trigger_type=ActionTrigger.get_value(self.base_data["triggerType"]),
-            service_type=ActionService.get_value(self.base_data["serviceType"]),
-            target_type=ActionTarget.get_value(self.base_data["targetType"]),
-        )(registration)
+        self.mock_register(self.base_data)(registration)
 
         data = {
             **self.base_data,
             "projects": [p.slug for p in self.projects],
         }
-
-        assert not registration.validate_action.called
+        assert not validate_action.called
```

Comment:
ah yeah this is a fun one

mypy sees things as immutable so:

```
assert foo.bar
...
assert not foo.bar
...  # mypy thinks this is impossible because the assertion above "must" have failed!
```

the easier fix for this is to use `assert_called()` and `assert_not_called()`